anesthetic/plot.py

```diff
@@ -539,27 +544,28 @@ def kde_plot_1d(ax, data, *args, **kwargs):
     q = kwargs.pop('q', 5)
     q = quantile_plot_interval(q=q)
 
-    x, w = sample_compression_1d(data, weights, ncompress)
-    kde = gaussian_kde(x, weights=w, bw_method=bw_method)
+    data_compressed, w = sample_compression_1d(data, weights, ncompress)
+    kde = gaussian_kde(data_compressed, weights=w, bw_method=bw_method)
+    xmin = quantile(data, q[0], weights)
+    xmax = quantile(data, q[-1], weights)
+    x = np.linspace(xmin, xmax, nplot)
+
     p = kde(x)
     p /= p.max()
-    i = ((x > quantile(x, q[0], w)) & (x < quantile(x, q[-1], w)))
     bw = np.sqrt(kde.covariance[0, 0])
-    pp = cut_and_normalise_gaussian(x[i], p[i], bw=bw,
-                                    xmin=data.min(), xmax=data.max())
+    pp = cut_and_normalise_gaussian(x, p, bw, xmin=data.min(), xmax=data.max())
     pp /= pp.max()
-    area = np.trapz(x=x[i], y=pp) if density else 1
-    ans = ax.plot(x[i], pp/area, color=color, *args, **kwargs)
+    area = np.trapz(x=x, y=pp) if density else 1
+    ans = ax.plot(x, pp/area, color=color, *args, **kwargs)
 
     if facecolor and facecolor not in [None, 'None', 'none']:
         if facecolor is True:
             facecolor = color
-        c = iso_probability_contours_from_samples(pp, contours=levels,
-                                                  weights=w)
+        c = iso_probability_contours_from_samples(pp, contours=levels)
```

Comment:
This should be iso_probability_contours, since they are no longer samples

Reply:
Done in [2a04ff3](https://github.com/williamjameshandley/anesthetic/pull/216/commits/2a04ff36263a4667f2028bdae25ba3fe31b7ee8b).

I've also taken away the option to not provide weights to `iso_probability_contours_from_samples`. This should help emphasize that this function is only meant for weighted samples.

Reply:
This isn't quite correct -- you can provide unweighted samples. The issue is that the plotting points are not necessarily samples, so you need a different algorithm to compute the iso probability contours.

Reply:
Reverted